File: fast_sync_dart_client/fast_sync_client/lib/src/absraction/service/isync_config.dart
import 'package:fast_sync_client/fast_sync_client.dart';

abstract class ISyncConfiguration {
  void setSyncableObject<T extends ISyncableObject>(String entityType,
      Function fromJson, Function toJson, ISyncableRepository<T> repository);
  void setObjectRepository<T extends ISyncableObject>(
      String entityType, ISyncableRepository<T> repository);
  ISyncableRepository<T> getObjectRepository<T extends ISyncableObject>(
      String type);
  ISyncManager getSyncManager();
  IhttpManager getHttpManager();
  void setHttpManager(IhttpManager httpManager);
  void setSyncVersionManager(SyncVersionManager syncVersionManager);
  SyncVersionManager getSyncVersionManager();
}

File: fast-sync-typescript-client-sdk/packages/fast-sync-client/src/abstraction/service/isync_config.ts
import { SyncVersionManager } from "../../implementation/service/sync_version_manager";
import { ISyncableRepository } from "../data/isyncable_repository";
import { ISyncableObject } from "../metadata/isyncable_object";
import { IHttpManager } from "./ihttp_manager";
import { ISyncManager } from "./isync_manager";

export interface ISyncConfiguration {
  setSyncableObject<T extends ISyncableObject>(
    entityType: string,
    repository: ISyncableRepository<T>
  ): void;

  setObjectRepository<T extends ISyncableObject>(
    entityType: string,
    repository: ISyncableRepository<T>
  ): void;

  getObjectRepository<T extends ISyncableObject>(
    type: string
  ): ISyncableRepository<T>;

  getSyncManager(): ISyncManager;

  getHttpManager(): IHttpManager;

  setHttpManager(httpManager: IHttpManager): void;

  setSyncVersionManager(syncVersionManager: SyncVersionManager): void;

  getSyncVersionManager(): SyncVersionManager;
}
